### PR TITLE
[exporter/datadog] Set gzip compression for native metrics export

### PR DESCRIPTION
**Pull request by songy23**
**Description:** <Describe what has changed.>
Set gzip compression for Datadog native metrics export

### CODE REVIEW

**Comment by songy23** [comment]
Please add `Skip changelog` since this isn't a user-facing change.

File: exporter/datadogexporter/metrics_exporter.go [hunk 1]
```diff
@@ -226,7 +226,8 @@ func (exp *metricsExporter) PushMetricsData(ctx context.Context, md pmetric.Metr
 			exp.params.Logger.Debug("exporting native Datadog payload", zap.Any("metric", ms))
 			_, experr := exp.retrier.DoWithRetries(ctx, func(context.Context) error {
 				ctx = clientutil.GetRequestContext(ctx, string(exp.cfg.API.Key))
-				_, httpresp, merr := exp.metricsAPI.SubmitMetrics(ctx, datadogV2.MetricPayload{Series: ms})
+				param := datadogV2.NewSubmitMetricsOptionalParameters().WithContentEncoding(datadogV2.METRICCONTENTENCODING_GZIP)
```
Comment (dineshg13) [inline exporter/datadogexporter/metrics_exporter.go:229]: Can we make this const ?

Comment (gbbr) [reply exporter/datadogexporter/metrics_exporter.go:229]: Why are you asking that? It would be quite an uncommon pattern.

Comment (dineshg13) [reply exporter/datadogexporter/metrics_exporter.go:229]: why create a variable for each request, that would be garbage collected .

Comment (songy23) [reply exporter/datadogexporter/metrics_exporter.go]: Good point, updated.

**Review by dineshg13 — APPROVED** [review]
LTGM

**Comment by dineshg13** [comment]
@songy23 wouldn't the change impact the egress for customer ? I would prefer to have changelog for same.

**Comment by songy23** [comment]
OK makes sense, added a changelog entry.

**Review by mx-psi — APPROVED** [review]

**Pull request merged**